adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DefaultDeviceResolver.java

```diff
@@ -98,14 +98,14 @@ public DefaultDeviceResolver(
             final Tracer tracer,
             final String adapterName,
             final CoapAdapterProperties config,
-            final CredentialsClientFactory credentialsClientFactory,
+            final CredentialsClient credentialsClientFactory,
             final TenantClient tenantClient) {
 
         this.context = Objects.requireNonNull(vertxContext);
         this.tracer = Objects.requireNonNull(tracer);
         this.adapterName = Objects.requireNonNull(adapterName);
         this.config = Objects.requireNonNull(config);
-        this.credentialsClientFactory = Objects.requireNonNull(credentialsClientFactory);
+        this.credentialsClient = Objects.requireNonNull(credentialsClientFactory);
         this.tenantClient = tenantClient;
```

Comment:
Missing _null_ check for the `tenantClient` and as per the JavaDoc it should also be _null_ checked. This is an existing issue and may be this could also be fixed here in this related PR.